src/ucp/tag/rndv.c

```diff
@@ -29,8 +29,9 @@ static int ucp_tag_rndv_is_get_op_possible(ucp_ep_h ep, uct_rkey_t rkey)
 
 static void ucp_rndv_rma_request_send_buffer_dereg(ucp_request_t *sreq)
 {
-    if ((UCP_DT_IS_CONTIG(sreq->send.datatype)) &&
-        (ucp_ep_is_rndv_lane_present(sreq->send.ep))) {
+    if (UCP_DT_IS_CONTIG(sreq->send.datatype) &&
+        ucp_ep_is_rndv_lane_present(sreq->send.ep) &&
+        ucp_request_is_send_buffer_reg(sreq) /* TODO: How is this possible? */) {
```

Comment:
what do you mean?
ucp_rndv_prepare_zcopy_send_buffer
and
ucp_rndv_rma_request_send_buffer_dereg
are not called one after the other, only one of them is called.

also, how does MM register the buffer if neither posix nor sysv have a reg func? MM shouldn't be chosen for rndv